packages/react-native-executorch/common/rnexecutorch/models/text_to_image/UNet.cpp

```diff
@@ -0,0 +1,56 @@
+#include "UNet.h"
+
+#include <executorch/extension/tensor/tensor_ptr_maker.h>
+
+namespace rnexecutorch::models::text_to_image {
+
+using namespace executorch::extension;
+
+UNet::UNet(const std::string &modelSource, int32_t modelImageSize,
+           int32_t numChannels, std::shared_ptr<react::CallInvoker> callInvoker)
+    : BaseModel(modelSource, callInvoker), modelImageSize(modelImageSize),
+      numChannels(numChannels) {}
+
+std::vector<float> UNet::generate(const std::vector<float> &latents,
+                                  int32_t timestep,
+                                  const std::vector<float> &embeddings) {
+  int32_t latentsImageSize = std::floor(modelImageSize / 8);
```

Comment:
Could you add the same comment about 8 with 3-down-sampling layers here as well?